app/services/oidc_jwt_token_validate.rb

```diff
@@ -0,0 +1,56 @@
+class OidcJwtTokenValidate
+  attr_reader :decoded_token
+  delegate :logger, to: :Rails
+
+  def initialize(jwt_token)
+    @jwt_token = jwt_token
+  end
+
+  def valid_jwt?
+    valid_signature? && type_bearer? && valid_aud?
+  rescue JWT::InvalidAudError
+    logger.error "OidcJwtToken SSO: Invalid token."
+    false
+  rescue JWT::ImmatureSignature
+    logger.error "OidcJwtToken SSO: JWT is for the future."
+    false
+  rescue JWT::ExpiredSignature
+    logger.error "OidcJwtToken SSO: Signature has expired."
+    false
+  rescue JWT::DecodeError
+    logger.error "OidcJwtToken SSO: Failed to decode JWT."
+    false
+  end
+
+  private
+
+  def generate_jwks_uri
```

Comment:
The method name is misleading as it does not generate a URI, but actually returns a key.